packages/commons/src/auth/authenticationMiddleware.ts

```diff
@@ -42,6 +42,13 @@ export const authenticationMiddleware: () => ZodiosRouterContextRequestHandler<E
         }
 
         const authData: AuthData = readAuthDataFromJwtToken(jwtToken);
+        const context = getMutableContext();
+        if (context) {
+          // eslint-disable-next-line functional/immutable-data
+          context.authData = authData;
+        } else {
+          logger.warn("Cannot set authData in context - context not found.");
```

Comment:
is this a critical error?

Reply:
I would not consider it critical: it is only related to the logging context - the worst that can happen is that we do not log the correlation ID. Still, it is unexpected: this should never happen because at this point the context is already initialized - it cannot be undefined, but the API of `globalStore.getStore()` returns a nullable value. Do you think we should log it as an error and not a warning?

Reply:
my concern is that in this point we are adding auth data to the context.
If the context is null, auth data will not be added to the context, but the auth middleware does not fail, which leads to unexpected (and unsecure?) result.

Does it make sense?

Reply:
@MalpenZibo / @galales now it shoul be clearer and more readable